drivers/espi/espi_npcx.c

```diff
@@ -0,0 +1,870 @@
+/*
+ * Copyright (c) 2020 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nuvoton_npcx_espi
+
+#include <assert.h>
+#include <drivers/espi.h>
+#include <drivers/gpio.h>
+#include <drivers/clock_control.h>
+#include <dt-bindings/espi/npcx_espi.h>
+#include <kernel.h>
+#include <soc.h>
+#include "espi_utils.h"
+#include "soc_miwu.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(espi, CONFIG_ESPI_LOG_LEVEL);
+
+struct espi_npcx_config {
+	uintptr_t base;
+	/* clock configuration */
+	struct npcx_clk_cfg clk_cfg;
+	/* mapping table between eSPI reset signal and wake-up input */
+	struct npcx_wui espi_rst_wui;
+	/* pinmux configuration */
+	const uint8_t   alts_size;
+	const struct npcx_alt *alts_list;
+};
+
+struct espi_npcx_data {
+	sys_slist_t callbacks;
+	uint8_t plt_rst_asserted;
+	uint8_t espi_rst_asserted;
+	uint8_t sx_state;
+#if defined(CONFIG_ESPI_OOB_CHANNEL)
+	struct k_sem oob_rx_lock;
+#endif
+};
+
+/* Driver convenience defines */
+#define DRV_CONFIG(dev) ((const struct espi_npcx_config *)(dev)->config)
+
+#define DRV_DATA(dev) ((struct espi_npcx_data *)(dev)->data)
+
+#define HAL_INSTANCE(dev) (struct espi_reg *)(DRV_CONFIG(dev)->base)
+
+/* eSPI channels */
+#define NPCX_ESPI_CH_PC              0
+#define NPCX_ESPI_CH_VW              1
+#define NPCX_ESPI_CH_OOB             2
+#define NPCX_ESPI_CH_FLASH           3
+#define NPCX_ESPI_CH_COUNT           4
+#define NPCX_ESPI_HOST_CH_EN(ch)     (ch + 4)
+
+/* eSPI max supported frequency */
+#define NPCX_ESPI_MAXFREQ_20         0
+#define NPCX_ESPI_MAXFREQ_25         1
+#define NPCX_ESPI_MAXFREQ_33         2
+#define NPCX_ESPI_MAXFREQ_50         3
+
+/* Minimum delay before acknowledging a virtual wire */
+#define NPCX_ESPI_VWIRE_ACK_DELAY    10ul /* 10 us */
+
+/* OOB channel maximum payload size */
+#define NPCX_ESPI_OOB_MAX_PAYLOAD    64
+#define NPCX_OOB_RX_PACKAGE_LEN(hdr) (((hdr & 0xff000000) >> 24) | \
+							((hdr & 0xf0000) >> 8))
+
+/* eSPI cycle type field for OOB and FLASH channels */
+#define ESPI_FLASH_READ_CYCLE_TYPE   0x00
+#define ESPI_FLASH_WRITE_CYCLE_TYPE  0x01
+#define ESPI_FLASH_ERASE_CYCLE_TYPE  0x02
+#define ESPI_OOB_GET_CYCLE_TYPE      0x21
+#define ESPI_OOB_TAG                 0x00
+#define ESPI_OOB_MAX_TIMEOUT         500ul /* 500 ms */
+
+/* eSPI bus interrupt configuration structure and macro function */
+struct espi_bus_isr {
+	uint32_t status_bit; /* bit in ESPISTS register */
+	uint32_t int_en_bit; /* bit in ESPIIE register */
+	uint32_t wake_en_bit; /* bit in ESPIWE register */
+	void (*bus_isr)(const struct device *dev); /* eSPI bus ISR */
+};
+
+#define NPCX_ESPI_BUS_INT_ITEM(event, isr) {     \
+	.status_bit = NPCX_ESPISTS_##event,      \
+	.int_en_bit = NPCX_ESPIIE_##event##IE,   \
```

Comment:
I'd like to keep the register bit name as the same as our datasheet. Or it will hard to find it in the datasheet and divergence issue with our HAL.

Reply:
This is just putting an underscore between a number and text. Since this is a macro anyway, you won't be able to look up this up in the datasheet since 3IE doesn't exist in code anywhere.

Either way, it is not that big of a deal, so you can leave it the way it is; it just seems like a more consistent name with that extra underscorce

Reply:
Thanks. What I mean it's the result of this macro is mapping to register bit name in "reg_def.h". (ie. NPCX_ESPIIE_CFGUPDIE -is composed by Chip family + register name + register bit/field name)  If we want to apply this, it also need to modify the register bit name in reg_def.h.